IntroToCSharp/Shared/Service/UserService.cs

```diff
@@ -236,6 +236,31 @@ public bool GiveXPAndGold(int xpToGive, int goldToGive)
     /// <returns>False if user is not logged in. Returns true otherwise.</returns>
     public bool GiveGold(int goldToGive) => GiveXPAndGold(0, goldToGive);
 
+    /// <summary>
+    /// Increases the counter for the amount of Gold Spent.
+    /// </summary>
+    /// <param name="goldToGive"></param>
+    /// <returns></returns>
+    public bool IncreaseGoldSpent(int goldToGive)
+    {
+        if (!_userData.IsLoggedIn) return false;
+        UserStats newStats = new(_userStats.XP, _userStats.Gold, _userStats.GoldAcquired, _userStats.GoldSpent + goldToGive);
+        _userData.UserStatsRef?.Set(newStats);
+        return true;
+    }
```

Comment:
Would it make sense to have a single `GiveGold` and check if the amount is positive or negative and apply the proper calculation?

e.g. When positive, increase `GoldAcquired`. When negative, increase `GoldSpent`

What are your thoughts?

Reply:
I think I was able to do what you were asking. Would you mind verifying?
@jcollard